pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/DefaultAggregationExecutor.java

```diff
@@ -112,40 +105,48 @@ public void aggregate(int[] docIdSet, int startIndex, int length) {
    *
    * @param aggrFuncContext aggregation function context.
    * @param resultHolder result holder.
-   * @param length length of the block.
    */
-  private void aggregateColumn(AggregationFunctionContext aggrFuncContext, AggregationResultHolder resultHolder,
-      int length) {
+  @SuppressWarnings("ConstantConditions")
+  private void aggregateColumn(ProjectionBlock projectionBlock, AggregationFunctionContext aggrFuncContext,
+      AggregationResultHolder resultHolder) {
     AggregationFunction aggregationFunction = aggrFuncContext.getAggregationFunction();
     String[] aggrColumns = aggrFuncContext.getAggregationColumns();
     String aggrFuncName = aggregationFunction.getName();
 
     Preconditions.checkState(aggrColumns.length == 1);
     String aggrColumn = aggrColumns[0];
+
+    ProjectionBlockValSet blockValueSet = null;
+    if (!aggrFuncName.equals(AggregationFunctionFactory.COUNT_AGGREGATION_FUNCTION)) {
+      Block dataBlock = projectionBlock.getDataBlock(aggrColumn);
+      blockValueSet = (ProjectionBlockValSet) dataBlock.getBlockValueSet();
```

Comment:
I am not fully sure if we should add some of the apis on ProjectionBlockValSet (eg getHashCodeArray) into the BlockValSet interface as they would likely not make sense outside of projection block. I kept these api's in ProjectionBlockValSet for efficiency, if we take them out, then we need to re-iterate over values and further process them (eg compute hash codes). So this boils down to performance v.s. clean interface.